crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -163,283 +163,285 @@ impl<C: Comments> ServerActions<C> {
         (is_action_fn, is_cache_fn)
     }
 
-    fn maybe_hoist_and_create_proxy(
+    fn maybe_hoist_and_create_proxy_for_server_action_arrow_expr(
         &mut self,
         ids_from_closure: Vec<Name>,
-        function: Option<&mut Box<Function>>,
-        arrow: Option<&mut ArrowExpr>,
-    ) -> Option<Box<Expr>> {
+        arrow: &mut ArrowExpr,
+    ) -> Box<Expr> {
         let action_name: JsWord = gen_action_ident(&mut self.reference_index);
-        let action_ident = Ident::new(action_name.clone(), DUMMY_SP, self.private_ctxt);
-        let export_name: JsWord = action_name;
 
         self.has_action = true;
-        self.export_actions.push(export_name.to_string());
+        self.export_actions.push(action_name.to_string());
 
-        if let Some(a) = arrow {
-            let register_action_expr = annotate_ident_as_server_reference(
-                action_ident.clone(),
-                ids_from_closure
-                    .iter()
-                    .cloned()
-                    .map(|id| Some(id.as_arg()))
-                    .collect(),
-                generate_action_id(
-                    &self.config.hash_salt,
-                    &self.file_name,
-                    export_name.to_string().as_str(),
-                ),
-            );
+        let action_ident = Ident::new(action_name.clone(), DUMMY_SP, self.private_ctxt);
```

Comment:
In a future PR I will create different spans for `action_ident`, based on whether we're dealing with a `Function` or an `ArrowExpr`. Having two different functions with well-defined params will make this change straight-forward.